tests/unit/test_experiments/test_simulation_with_experiment.py

```diff
@@ -261,77 +261,84 @@ def test_solve_with_sensitivities_and_experiment(self):
+                solver=solver,
+                parameter_values=param,
+            )
+            solution = sim.solve(
+                inputs={input_param_name: input_param_value},
+                calculate_sensitivities=calculate_sensitivities,
+            )
+            solutions.append(solution)
 
-        # check solutions are the same, leave out the last solution point as it is slightly different
-        # for each solve due to numerical errors
-        # TODO: scipy solver does not work for this experiment, with or without sensitivities,
-        # so we skip this test for now
-        for i in range(1, len(solutions) - 2):
-            np.testing.assert_allclose(
-                solutions[0]["Voltage [V]"].data[:-1],
-                solutions[i]["Voltage [V]"](solutions[0].t[:-1]),
-                rtol=5e-2,
-                equal_nan=True,
+        model = pybamm.lithium_ion.SPM()
+        param = model.default_parameter_values
+        base_input_param_value = param[input_param_name]
+        fd_tol = 1e-4
+        for dh in [-fd_tol, fd_tol]:
+            model = pybamm.lithium_ion.SPM()
+            param = model.default_parameter_values
+            input_param_value = base_input_param_value * (1.0 + dh)
+            param.update({input_param_name: "[input]"})
+            sim = pybamm.Simulation(
+                model,
+                experiment=experiment_2step,
+                solver=solver,
+                parameter_values=param,
+            )
+            solution = sim.solve(
+                inputs={input_param_name: input_param_value},
             )
+            solutions.append(solution)
 
-        # check sensitivities are roughly the same. Sundials isn't doing error control on the sensitivities
-        # by default, and the solution can be quite coarse for quickly changing sensitivities
-        sens_casadi = (
-            solutions[1]["Voltage [V]"]
-            .sensitivities[input_param_name][:-2]
-            .full()
-            .flatten()
+        # check solutions are the same, leave out the last solution point as it is slightly different
+        # for each solve due to numerical errors
+        np.testing.assert_allclose(
+            solutions[0]["Voltage [V]"].data[:-1],
+            solutions[1]["Voltage [V]"](solutions[0].t[:-1]),
+            rtol=5e-2,
+            equal_nan=True,
         )
+
+        # use finite difference to check sensitivities
+        t = solutions[0].t[:-2]
+        soln_neg = solutions[2]["Voltage [V]"](t)
+        soln_pos = solutions[3]["Voltage [V]"](t)
+        sens_fd = (soln_pos - soln_neg) / (2 * fd_tol * base_input_param_value)
         sens_idaklu = np.interp(
-            solutions[1].t[:-2],
-            solutions[3].t,
-            solutions[3]["Voltage [V]"]
+            t,
+            solutions[1].t,
+            solutions[1]["Voltage [V]"]
             .sensitivities[input_param_name]
             .full()
             .flatten(),
         )
```

Comment:
Can we change this to a higher order interpolator

Reply:
I tried this but it doesn't work very well due to the discontinuities in the experiment, linear interp behaves better